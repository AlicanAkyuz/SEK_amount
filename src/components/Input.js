import React from 'react';
import PropTypes from 'prop-types';

import { FullWidthContainer, PrimaryText, InputField } from '../containers/Main/styles';

const Input = ({
  title, type, value, placeholder, onchange,
}) => (
  <FullWidthContainer>
    <PrimaryText>{title}</PrimaryText>
    <InputField type={type} value={value} placeholder={placeholder} onChange={e => onchange(e)} />
  </FullWidthContainer>
);

Input.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  placeholder: PropTypes.string.isRequired,
  onchange: PropTypes.func.isRequired,
};

export default Input;
