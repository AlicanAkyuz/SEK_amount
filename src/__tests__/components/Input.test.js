import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Input from '../../components/Input';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Input
      title="Sek Amount"
      type="text"
      value="SEK"
      placeholder="Sek Amount"
      onchange={e => e.target.value}
    />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('correctly renders 5 props', () => {
  const tree = renderer
    .create(
      <Input
        title="Sek Amount"
        type="text"
        value="SEK"
        placeholder="Sek Amount"
        onchange={e => e.target.value}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('handles both string and number value types', () => {
  const tree = renderer
    .create(
      <Input
        title="Sek Amount"
        type="text"
        value={15}
        placeholder="Sek Amount"
        onchange={e => e.target.value}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
