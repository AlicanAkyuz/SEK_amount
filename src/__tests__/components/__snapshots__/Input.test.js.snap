// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders 5 props 1`] = `
<div
  className="css-1a821q7"
>
  <h3
    className="css-qwi5oq"
  >
    Sek Amount
  </h3>
  <input
    className="css-dbs8c2"
    onChange={[Function]}
    placeholder="Sek Amount"
    type="text"
    value="SEK"
  />
</div>
`;

exports[`handles both string and number value types 1`] = `
<div
  className="css-1a821q7"
>
  <h3
    className="css-qwi5oq"
  >
    Sek Amount
  </h3>
  <input
    className="css-dbs8c2"
    onChange={[Function]}
    placeholder="Sek Amount"
    type="text"
    value={15}
  />
</div>
`;
